metric_providers/disk/io/cgroup/container/source.c

```diff
@@ -0,0 +1,189 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <errno.h>
+#include <unistd.h>
+#include <sys/time.h>
+#include <string.h> // for strtok
+#include <getopt.h>
+
+typedef struct container_t { // struct is a specification and this static makes no sense here
+    char path[BUFSIZ];
+    char *id;
+} container_t;
+
+typedef struct disk_io_t { // struct is a specification and this static makes no sense here
+    long int rbytes;
+    long int wbytes;
+} disk_io_t;
+
+
+// All variables are made static, because we believe that this will
+// keep them local in scope to the file and not make them persist in state
+// between Threads.
+// in any case, none of these variables should change between threads
```

Comment:
Are you using threads?

Reply:
there was an idea to do so at the time of writing, thus the warning commentary. atm we don't though.